File: Lab 4/Tutor Demo-ZWordCount/ZWordCount.java
public class ZWordCount {
    public static void main(String[] args) {
        String[] sentences = {
            "green ideas Zleep furiously",
            "a zebra graZed",
            "zorro WAZ zore",
            "I feel snoozy"
        };
        for (String sentence : sentences) {
            System.out.println(sentence);
        }
        System.out.println("Words containing a 'z' = " + countZ(sentences));
    }

    public static int countZ(String[] sentences) {
        int sum = 0;
        for(String sentence : sentences){
            sum += countZ(sentence);
        }
        return sum;
    }

    public static int countZ(String sentence) {
        String[] words = sentence.split(" +"); // The + means one OR MORE spaces
        int count = 0;
        for(String word : words){
            if(containsZ(word)){ // can talk about redundant == true
                count++;
            }
        }
        return count;
    }

    public static boolean containsZ(String word) {
        return word.contains("z") || word.contains("Z");
    }
}

File: Lab 4/Tutor Demo-ZWordCount/ZWordCount.py
def containsZ(word):
    return ('z' in word)

# Have to give different name for functions in python
def zWord(sentence):
    count = 0
    # split function in python defaultly ignore all space
    # (' ') can only split one space
    # (' +') doesn't work in python
    for word in sentence.split():
        if containsZ(word):
            count = count + 1
    return count

def zWords(sentences):
    sum1 = 0
    for sentence in sentences:
        sum1 += zWord(sentence.lower())
    return sum1

if __name__ == "__main__":
    sentences = ["green ideas Zleep furiously",
                 "a zebra graZed",
                 "zorro WAZ zore",
                 "I feel snoozy"]
    for sentence in sentences:
        print(sentence)
    
    print("Words containing a 'z' = %d" %zWords(sentences))
